src/components/Trade/TradeInput.tsx

```diff
@@ -257,8 +257,17 @@ export const TradeInput = () => {
     [handleInputChange],
   )
 
-  const handleSellAccountIdChange: AccountDropdownProps['onChange'] = accountId =>
+  const handleSellAccountIdChange: AccountDropdownProps['onChange'] = accountId => {
     setValue('selectedSellAssetAccountId', accountId)
+    if (!swapperSupportsCrossAccountTrade && bestTradeSwapper) {
+      /*
+        This is extremely dangerous. We only want to do this if we have a swapper, and that swapper does not do either of:
+          - Trades between assets on the same chain but different accounts
+          - Trades between assets on different chains (and possibly different accounts)
+       */
+      setValue('selectedBuyAssetAccountId', accountId)
```

Comment:
We now do the dangerous `selectedBuyAssetAccountId` sync here, and be very clear about what we are doing, and why.